scripts/components/RepairRequestForm.jsx

```diff
@@ -50,55 +55,62 @@ var RepairRequestForm = React.createClass({
         // Clear the form
         this.setState({
           description: '',
-          dataUri: '',
+          image: '',
         });
 
         this.props.repairRequestAdded();
+        this.refs.dialog.dismiss();
       }
     });
   },
 
-  /**
-   * Callback for a user selecting a file for upload. Reads the file as a base64
-   * encoded data URI and stores this in component state.
-   */
-  onFileSelected(event) {
-    var reader = new FileReader(); // File API
-    var file = event.target.files[0];
+  onImageSelected(payload) {
+    this.setState({ image: payload.dataURL });
+  },
 
-    reader.onload = upload => this.setState({ dataUri: upload.target.result });
-    reader.readAsDataURL(file);
+  onImageSizeError(error) {
+    var file = error.file;
+    var sizeLimit = error.sizeLimit / 1000; // in KB (base10)
+    var error = `${file.name} exceeds size limit of ${sizeLimit}kb.`;
+    this.setState({ fileSizeError: error });
   },
 
   render() {
-    var { description, dataUri } = this.state;
+    var { description, fileSizeError } = this.state;
+    var sizeError = fileSizeError ? (
+      <ErrorMessage fillBackground={true}>Error: {fileSizeError}</ErrorMessage>
+    ) : null;
     var errorMessage;
+    var standardActions = [
+      { text: 'Cancel' },
+      { text: 'Lodge Repair Request', onTouchTap: this.onSubmit, ref: 'submit' }
+    ];
     return (
       <div style={style.formContainer}>
-        <Paper zDepth={1}>
-          <form style={style.form} onSubmit={this.onSubmit}>
-            <h3 style={style.heading}>Lodge a New Repair Request</h3>
-            <div style={style.error}> { errorMessage } </div>
-            <TextField
-              value={description}
-              multiLine={true}
-              name="Description"
-              onChange={this.onChange.bind(this, 'description')}
-              floatingLabelText="Description" />
-            <div style={style.inputContainer}>
-              {dataUri ?
-                (<img style={style.img} src={this.state.dataUri} />) :
-                (null)}
-              <input type="file" name="file" onChange={this.onFileSelected} />
-            </div>
-            <RaisedButton
-              type="submit"
-              label="Lodge Repair Request"
-              primary={true}
-              backgroundColor="#2ECC71"
-              style={style.button} />
-          </form>
-        </Paper>
+        <RaisedButton label="Lodge a New Repair Request"
+                      primary={true}
+                      onClick={this.onButtonClick} />
+        <Dialog
+          title="Lodge a New Repair Request"
+          actions={standardActions}
+          actionFocus="submit"
+          modal={this.state.modal}
+          ref="dialog">
+          <div style={style.error}> { errorMessage } </div>
+          <TextField
+            value={description}
+            multiLine={true}
```

Comment:
This field is intended to have a paragraph written in it. Can you use the `fullWidth` property so that it fills the available space for writing.

Also, similar to not using `Submit` to describe a button, we also shouldn't use `Description` to describe the purpose of a field where we want complex user input. We want to guide the user with using the form field correctly. A possible improvement would be:

`floatingLabelText="Describe what needs repairing, how urgent it is, and how the damage occurred."`